src/ondemand.coffee

```diff
@@ -231,119 +86,185 @@ module.exports = (robot) ->
         if err
           return robot.logger.error err
-        ws.getConclaveAll platform, (err, challenge) ->
-          if err 
-            res.send
-            return robot.logger.error err
-          if challenge != ''
-            res.send util.format(challengeFormat, dsUtil.codeMulti, challenge, dsUtil.blockEnd)
-          else
-            res.send noChallengeString
-    
+        checkWorldstate platform
+        setTimeout (-> res.send worldStates[platform].getConclaveAllString()), 300
     else if dailyRegExp.test(params)
       robot.logger.debug 'Entered daily challenge'
       userDB.getPlatform res.message.room, (err, platform) ->
         if err
           return robot.logger.error err
-        ws.getConclaveDailies platform, (err, challenge) ->
-          if err 
-            return robot.logger.error err
-          if challenge != ''
-            res.send util.format(challengeFormat, dsUtil.codeMulti, challenge, dsUtil.blockEnd)
-          else
-            res.send noChallengeString
-    
+        checkWorldstate platform
+        setTimeout (-> res.send worldStates[platform].getConclaveDailiesString()), 300   
     else if weeklyRegExp.test(params)
       robot.logger.debug 'Entered weekly challenge'
       userDB.getPlatform res.message.room, (err, platform) ->
         if err
           return robot.logger.error err
-        ws.getConclaveWeeklies platform, (err, challenge) ->
-          if err
-            return robot.logger.error err
-          if challenge != ''
-            res.send util.format(challengeFormat, dsUtil.codeMulti, challenge, dsUtil.blockEnd)
-          else
-            res.send noChallengeString
+        checkWorldstate platform
+        setTimeout (-> res.send worldStates[platform].getConclaveWeekliesString()), 300
+
     else
       robot.logger.debug 'Entered null challenge'
       conclaveInstructAll = 'conclave all - print all conclave challenges.'
       conclaveInstructWeekly = 'conclave weekly - print weekly conclave challenges.'
       conclaveInstructDaily = 'conclave daily - print conclave daily challenges.'
       res.send util.format('%sPossible uses include:%s%s%s%s%s%s', 
-                                    dsUtil.codeMulti, dsUtil.lineEnd, 
-                                    conclaveInstructAll, dsUtil.lineEnd, 
-                                    conclaveInstructWeekly, dsUtil.lineEnd, 
-                                    conclaveInstructDaily, dsUtil.blockEnd)
+                                    md.codeMulti, md.lineEnd, 
+                                    conclaveInstructAll, md.lineEnd, 
+                                    conclaveInstructWeekly, md.lineEnd, 
+                                    conclaveInstructDaily, md.blockEnd)
+  robot.respond /damage/, (res) ->  
+    damageURL = 'http://morningstar-wf.com/chart/Damage_2.0_Resistance_Flowchart.png'
+    res.send "${md.codeMulti}#{md.linkBegin}Damage 2.0#{md.linkMid}#{damageURL}#{md.linkEnd}#{md.blockEnd}"
```

Comment:
Indeed. I'll fix it. I hadn't had a problem on it because it apparently still works,  somehow, but I'll fix it.